frontend/public/components/utils/storage-class-dropdown.tsx

```diff
@@ -0,0 +1,192 @@
+          state.title = this.getTitle(item.kindLabel, item.name);
+        }
+      }
+
+      Object.keys(unorderedItems).sort().forEach(function(key) {
+        state.items[key] = unorderedItems[key];
+      });
+      this.setState(state);
+    }
+
+    shouldComponentUpdate(nextProps, nextState) {
+      return !_.isEqual(this.state, nextState);
+    }
+
+    autocompleteFilter = (text, item) => fuzzy(text, item.props.name);
+
+    getTitle = (kindLabel, name) => {
+      return kindLabel ? <ResourceName kind="StorageClass" name={name} /> : <span >{name}</span>;
+    }
+
+    onChange = key => {
+      const { name, kindLabel, accessMode } = _.get(this.state, ['items', key], {});
+      this.setState({
+        selectedKey: key,
+        title: this.getTitle(kindLabel, name),
+      });
+      this.props.onChange(key, accessMode);
+    };
+
+    render() {
+      const { placeholder, id, loaded } = this.props;
+      const items = {};
+      _.each(
+        this.state.items,
+        (props, key) =>
+          (items[key] = <StorageClassDropdownEntry {...props} />)
+      );
+
+      const { selectedKey, defaultClass } = this.state;
+
+      const Component = <Dropdown
+        autocompleteFilter={this.autocompleteFilter}
+        autocompletePlaceholder={placeholder}
+        items={items}
+        selectedKey={selectedKey}
+        title={this.state.title}
+        onChange={this.onChange}
+        id={id}
+        menuClassName="dropdown-menu--text-wrap"
+      />;
+      // Only show the dropdown if 'no storage class' is not the only option which depends on defaultClass
+      const itemsAvailableToShow = (_.size(items)>1 && !defaultClass) || (_.size(items)>0 && defaultClass);
+      const helpMes = defaultClass ? `Default storage class ${defaultClass} will be used if no other storage class selected.` : 'Optional storage class for the new claim.';
+      return <React.Fragment>
+        {loaded && itemsAvailableToShow &&
+          <div>
+            <label className="control-label" htmlFor="storageclass-dropdown">
+            Storage Class
+            </label>
+            {Component}
+            <p className="help-block">
+              {helpMes}
+            </p>
+          </div>
+        }
+      </React.Fragment>;
+    }
+}
+
+export const StorageClassDropdown = props => {
+  return <Firehose resources={[{ kind: 'StorageClass', prop: 'StorageClass', isList: true }]}>
+    <StorageClassDropdown_ {...props} />
+  </Firehose>;
+};
+
+const StorageClassDropdownEntry = props => {
+  const storageClassProperties = [props.default ? ' (default)' : '', props.description, props.accessMode, props.provisioner, props.type, props.zone];
+  const storageClassDescriptionLine = storageClassProperties.filter(Boolean).join(' | ');
+  return <span>
+    {props.kindLabel ? <ResourceName kind={props.kindLabel} name={props.name} /> : <span >{props.name}</span>}
+    <span className="text-muted co-form-subsection"> {storageClassDescriptionLine}</span>
```

Comment:
We should use flexbox columns to line up the description with the name exactly. This isn't really a `co-form-subsection` and the text doesn't align with this style.

Reply:
Can you give me an example of how to do that?  I am trying flexbox but the issue is the icon is the first element and flexbox aligns with the icon.  The only solution working for right now is physically moving the description line over with some pixel offset.

Reply:
You could use flexbox to layout two columns. First column is the icon only. Second column contains stacked divs with the name on top and description on bottom. (You wouldn't use ResourceName in this case.)

@rhamilto or @sg00dwin might have a better suggestion :)

Reply:
Ok, reworked the entries in the dropdown to use flexbox.  Here is a picture which shows the text lines up now on all the entires. @spadgett @cshinn @beanh66 
![storage-dropdown-entries](https://user-images.githubusercontent.com/18728857/49179703-16a8fd00-f310-11e8-9da3-21dd4c488a1f.png)